zcash_client_backend/src/data_api.rs

```diff
@@ -913,6 +920,78 @@ pub trait WalletRead {
     ) -> Result<HashMap<TransparentAddress, NonNegativeAmount>, Self::Error> {
         Ok(HashMap::new())
     }
+
+    /// Returns the metadata associated with a given transparent receiver in an account
+    /// controlled by this wallet.
+    ///
+    /// This is equivalent to (but may be implemented more efficiently than):
+    /// ```no_run
+    /// if let Some(result) = self.get_transparent_receivers(account)?.get(address) {
+    ///     return Ok(result.clone());
+    /// }
+    /// if let Some(result) = self.get_reserved_ephemeral_addresses(account, false)?.get(address) {
+    ///     return Ok(result.clone());
+    /// }
+    /// Ok(None)
+    /// ```
+    ///
+    /// Returns `Ok(None)` if the address is not recognized, or we do not have metadata for it.
```

Comment:
It is okay to conflate these two conditions because the metadata only consists of the scope and address index (and implicitly the account id), which will be known for any plausible data source implementation. In particular, even if the wallet supports tracking some FVKs and/or IVKs without having the spending key, we will still have metadata for every address that we can track.